new-frontiers-14/frontier-station-14
# Spent BBs are no longer reloadable, add RGBB gun.

## About the PR



Edits the prototypes for BBs.  Turns existing BBs into caseless cartridges that shoot spent BBs - the spent BBs now have the finite  lifetime, and cannot be reloaded into the rifle.

Adds an RGB version of BBs, makes a box of them available as an emaggable item available in the NonLethalVend, and adds them as a rare item in the mail table.  RGBBs do not have a point light until fired, when they will despawn in 30 seconds (or on most collisions).

## Why / Balance


Reloading a spent BB into the gun would eventually despawn it and prevent the gun from firing again.  Removing a BB from a box would cause it to eventually despawn.

RGBB?  Funny.

## How to test


1. Set the probability for MailNFRGBBGun to 9999.
2. Spawn, run mailnow.
3. You should get an mail with an empty BB gun and a box of RGBBs.
4. Cycle some RGBBs.  They should not light up.
5. Load an RGBB into the gun, it should load.
6. Shoot some RGBBs.  They should light up and despawn.
7. Grab a shot RGBB, try to load it back into the gun.  It should not load.
8. Spawn a box of regular BBs.
9. Cycle some BBs.  They should not light up.
10. Load some BBs into the gun.  They should load.
11. Shoot some BBs.  They should not light up and eventually despawn.
12. Grab a shot BB, try to load it back into the gun.  It should not load.

## Media


![image](https://github.com/user-attachments/assets/e567b340-8747-4ddf-a3f6-c77c13498e1f)

## Requirements


## Breaking changes


**Changelog**



:cl:
- fix: Shot BBs now do not load into a BB gun.
- add: RGBBs are now available from the mail and as an emagged item from the NonLethalVend.
